commit 8c157569d0f3dd01cf0138aa02dfba0a88d893d3
Author: Louis-Francis Ratté-Boulianne <lfrb@collabora.com>
Date:   Mon Sep 11 02:52:16 2023 -0400

    panfrost: Add method to get size of AFBC superblocks valid data
    
    The compute shader is going through all the AFBC header blocks and
    calculating the size of valid data by summing up the subblock
    sizes (and taking into account solid color, uncompressed mode and
    copy blocks). The result is written into the given buffer (array
    of `pan_afbc_block_info`).
    
    The size is rounded up to the alignment parameter directly in the
    shader.
    
    Signed-off-by: Louis-Francis Ratté-Boulianne <lfrb@collabora.com>
    Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/25012>
    [bero@lindev.ch: Port to Panthor]
    Signed-off-by: Bernhard Rosenkränzer <bero@lindev.ch>

diff --git a/src/gallium/drivers/panfrost/pan_afbc_cso.c b/src/gallium/drivers/panfrost/pan_afbc_cso.c
index 7ea74252b98..17bd04b4a59 100644
--- a/src/gallium/drivers/panfrost/pan_afbc_cso.c
+++ b/src/gallium/drivers/panfrost/pan_afbc_cso.c
@@ -295,7 +295,6 @@ panfrost_afbc_get_shaders(struct panfrost_context *ctx,
    }
 
    COMPILE_SHADER(size, key.bpp, key.align);
-   COMPILE_SHADER(pack, key.align, key.tiled);
 
 #undef COMPILE_SHADER
 
diff --git a/src/gallium/drivers/panfrost/pan_cmdstream.c b/src/gallium/drivers/panfrost/pan_cmdstream.c
index 1eab3425055..9c09ee24edf 100644
--- a/src/gallium/drivers/panfrost/pan_cmdstream.c
+++ b/src/gallium/drivers/panfrost/pan_cmdstream.c
@@ -1,4 +1,5 @@
 /*
+ * Copyright (C) 2023 Amazon.com, Inc. or its affiliates.
  * Copyright (C) 2018 Alyssa Rosenzweig
  * Copyright (C) 2020 Collabora Ltd.
  * Copyright © 2017 Intel Corporation
@@ -40,6 +41,7 @@
 
 #include "decode.h"
 
+#include "pan_afbc_cso.h"
 #include "pan_blend.h"
 #include "pan_blitter.h"
 #include "pan_bo.h"
diff --git a/src/gallium/drivers/panfrost/pan_resource.c b/src/gallium/drivers/panfrost/pan_resource.c
index db6673b750e..12f3fa18a63 100644
--- a/src/gallium/drivers/panfrost/pan_resource.c
+++ b/src/gallium/drivers/panfrost/pan_resource.c
@@ -4,6 +4,7 @@
  * Copyright (C) 2014-2017 Broadcom
  * Copyright (C) 2018-2019 Alyssa Rosenzweig
  * Copyright (C) 2019 Collabora, Ltd.
+ * Copyright (C) 2023 Amazon.com, Inc. or its affiliates
  *
  * Permission is hereby granted, free of charge, to any person obtaining a
  * copy of this software and associated documentation files (the "Software"),
